### DISCUSSION TITLE
Return results in sahi/predict.py

### DISCUSSION THREAD

**Discussion by imandrealombardo** [obss/sahi#1114] [Q&A] [answered]
Hi,

I noticed that the function [predict](https://github.com/obss/sahi/blob/c9469f5baf57d614eb36f1c8a063e38ec1c62146/sahi/predict.py#L342) doesn't return results like in [get_sliced_prediction](https://github.com/obss/sahi/blob/c9469f5baf57d614eb36f1c8a063e38ec1c62146/sahi/predict.py#L125). There, it returns `object_prediction_list`, a list of `sahi.prediction.ObjectPrediction`.

Would it be possible to have the `predict` function return a list of object predictions for each image in the batch?

**Comment by fcakyon** [comment, marked as answer]
@imandrealombardo you can set `export_pickle=True` https://github.com/obss/sahi/blob/1240bab3b838fd30929474debe9e086f15bd07af/sahi/predict.py#L369

and set `return_dict=True` https://github.com/obss/sahi/blob/1240bab3b838fd30929474debe9e086f15bd07af/sahi/predict.py#L381

then predict will return `{'export_dir': str}` then you can load the pickle in export dir to access `object_prediction_list` 👍🏻
